tests/integration/kubernetes/gha-run.sh

```diff
@@ -275,14 +275,15 @@ function run_tests() {
 	# and enable cri plugin in containerd config.
 	# TODO: enable testing auto-generated policy for other types of hosts too.
 
-	if [ "${KATA_HOST_OS}" = "cbl-mariner" ]; then
+	if [ "${KATA_HOST_OS}" = "cbl-mariner" ] || [ "${KATA_HYPERVISOR}" = "qemu-tdx" ] ; then
 
 		export AUTO_GENERATE_POLICY="yes"
-
-		# set default containerd config
-		sudo containerd config default | sudo tee /etc/containerd/config.toml > /dev/null
-		echo "containerd config has been set to default"
-		sudo systemctl restart containerd && sudo systemctl is-active containerd
+		if [ "${GENPOLICY_PULL_METHOD}" = "containerd" ]; then
```

Comment:
This block is tricky.  If one enable GENPOLICY_PULL_METHOD=containerd in a scenary where the test machine is the k8s worker, it will overwrite containerd's configuration that was already proper configured by the kata-deploy to run the tests. So I appreciate if you could add a comment/warn on this block explaining when `GENPOLICY_PULL_METHOD=containerd` is allowed.

Reply:
@wainersm , thanks for pointing out this confusion! I just did some clean-up in this area, in https://github.com/kata-containers/kata-containers/pull/9998 . We'll rebase this PR on top of that.